lib/matplotlib/rcsetup.py

```diff
@@ -700,6 +700,12 @@ def cycler(*args, **kwargs):
     return reduce(operator.add, (ccycler(k, v) for k, v in validated))
 
 
+class _DunderChecker(ast.NodeVisitor):
+    def visit_Attribute(self, node):
+        if node.attr.startswith("__") and node.attr.endswith("__"):
+            raise ValueError("cycler strings with dunders are forbidden")
```

Comment:
Actually you need to `self.generic_visit(node)` as well here to continue walking the tree (https://docs.python.org/3/library/ast.html#ast.NodeVisitor.generic_visit).  The approval is conditional on fixing that, sorry for missing it.

Reply:
done.

Reply:
I'm not having any luck at writing a string that fails this without this though.

Reply:
I found one...